switch: panic when there are no matching branches

### Steps to reproduce

- have only feature branches
- run `git town switch --type=contribution`

buggy behavior
- the list of branches is empty
- when hitting ENTER it crashes

### Command output

```shell
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/git-town/git-town/v16/internal/cli/dialog/components/list.List[...].SelectedEntry(...)
        /Users/kevlar/d/git-town/internal/cli/dialog/components/list/list.go:180
github.com/git-town/git-town/v16/internal/cli/dialog/components/list.List[...].SelectedData(...)
        /Users/kevlar/d/git-town/internal/cli/dialog/components/list/list.go:176
github.com/git-town/git-town/v16/internal/cli/dialog.SwitchBranch({0x1400026ad50, 0x0, 0x8?}, 0x0, 0x1, {0x103d9f340, 0x0, 0x0})
        /Users/kevlar/d/git-town/internal/cli/dialog/switch_branch.go:38 +0x374
github.com/git-town/git-town/v16/internal/cmd.executeSwitch({0x14000112570, 0x0, 0x1}, 0x0, 0x0, 0x0, {0x140001164e8, 0x1, 0x1})
        /Users/kevlar/d/git-town/internal/cmd/switch.go:72 +0x3fc
github.com/git-town/git-town/v16/internal/cmd.switchCmd.func1(0x14000228f08, {0x14000112570, 0x0, 0x1})
        /Users/kevlar/d/git-town/internal/cmd/switch.go:42 +0xd4
github.com/spf13/cobra.(*Command).execute(0x14000228f08, {0x14000112550, 0x1, 0x1})
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:985 +0x840
github.com/spf13/cobra.(*Command).ExecuteC(0x1400016a308)
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1117 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/kevlar/d/git-town/vendor/github.com/spf13/cobra/command.go:1041
github.com/git-town/git-town/v16/internal/cmd.Execute()
        /Users/kevlar/d/git-town/internal/cmd/core.go:40 +0x3c8
main.main()
        /Users/kevlar/d/git-town/main.go:18 +0x28
```


### Additional information

It should exit with a message saying that there are no branches to switch to.

--- a/features/switch/no_branches.feature
+++ b/features/switch/no_branches.feature
@@ -1,0 +1,15 @@
+Feature: switch branches
+
+  Scenario: no branches to switch to
+    Given a Git repo with origin
+    And the branches
+      | NAME  | TYPE    | PARENT | LOCATIONS |
+      | alpha | feature | main   | local     |
+    And the current branch is "alpha"
+    When I run "git-town switch zonk"
+    Then it runs no commands
+    And it prints the error:
+      """
+      no branches to switch to
+      """
+    And the current branch is still "alpha"

--- a/internal/cmd/switch.go
+++ b/internal/cmd/switch.go
@@ -68,7 +68,7 @@ func executeSwitch(args []string, allBranches configdomain.AllBranches, verbose 
 	branchesAndTypes := repo.UnvalidatedConfig.Config.Value.BranchesAndTypes(data.branchNames)
 	defaultBranchType := repo.UnvalidatedConfig.Config.Value.DefaultBranchType
 	entries := SwitchBranchEntries(data.branchesSnapshot.Branches, branchTypes, branchesAndTypes, data.config.Config.Lineage, defaultBranchType, allBranches, data.regexes)
-	if len(entries) < 2 {
+	if len(entries) == 0 {
 		return errors.New(messages.SwitchNoBranches)
 	}
 	cursor := SwitchBranchCursorPos(entries, data.initialBranch)

--- a/internal/cmd/switch.go
+++ b/internal/cmd/switch.go
@@ -68,6 +68,9 @@ func executeSwitch(args []string, allBranches configdomain.AllBranches, verbose 
 	branchesAndTypes := repo.UnvalidatedConfig.Config.Value.BranchesAndTypes(data.branchNames)
 	defaultBranchType := repo.UnvalidatedConfig.Config.Value.DefaultBranchType
 	entries := SwitchBranchEntries(data.branchesSnapshot.Branches, branchTypes, branchesAndTypes, data.config.Config.Lineage, defaultBranchType, allBranches, data.regexes)
+	if len(entries) < 2 {
+		return errors.New(messages.SwitchNoBranches)
+	}
 	cursor := SwitchBranchCursorPos(entries, data.initialBranch)
 	branchToCheckout, exit, err := dialog.SwitchBranch(entries, cursor, data.uncommittedChanges, data.dialogInputs.Next())
 	if err != nil || exit {

--- a/internal/messages/en.go
+++ b/internal/messages/en.go
@@ -215,6 +215,7 @@ nd will be removed in future versions of Git Town.`
 	SquashCommitAuthorSelection    = "Selected squash commit author: %s\n"
 	SquashMessageProblem           = "cannot comment out the squash commit message: %w"
 	StatusFileNotFound             = "No status file found for this repository."
+	SwitchNoBranches               = "no branches to switch to"
 	SwitchUncommittedChanges       = "uncommitted changes\n"
 	SyncFeatureBranches            = "Sync feature branches: %s\n"
 	SyncPerennialBranches          = "Sync perennial branches: %s\n"

